lms/djangoapps/mobile_api/users/serializers.py

```diff
@@ -34,10 +37,20 @@ def to_native(self, course_overview):
                 kwargs={'course_id': course_id},
                 request=request
             )
+            try:
+                get_discussion_course_or_404(CourseKey.from_string(course_id), request.user)
```

Comment:
There are two issues with this approach:
1. (Easy) This code should not call a helper that raises a 404.  Instead, it should call `is_discussion_enabled` directly. 
2. (Harder) For performance reasons, we intentionally do not get the `course module` in this endpoint.  The way the edx-platform is currently implemented, getting the `course_module` object for every course a user is enrolled in (for presenting to the dashboard) is slow.  So instead, we get the `course_overview` object for each course.